contrib/cloudevents/sdkgo.v2/cloudevents.go

```diff
@@ -0,0 +1,149 @@
+package sdkgov2
+
+import (
+	"context"
+
+	"github.com/DataDog/dd-trace-go/v2/ddtrace/ext"
+	"github.com/DataDog/dd-trace-go/v2/ddtrace/tracer"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/event"
+)
+
+// Option is a functional option for configuring CloudEvents tracing.
+type Option func(*config)
+
+type config struct {
+	includeSubject bool
+}
+
+func defaultConfig() *config {
+	return &config{
+		includeSubject: false,
+	}
+}
+
+// WithSubject enables inclusion of the event subject in the span tags.
+// Note: Event subjects may contain sensitive data, so this is opt-in.
+func WithSubject() Option {
+	return func(c *config) {
+		c.includeSubject = true
+	}
+}
+
+// TraceWrapCloudEventsHandler wraps a CloudEvents handler to enable distributed tracing.
+// It extracts trace context from the event, creates a consumer span, and propagates the
+// context to the handler. The resourceName is used as the span's resource name.
+//
+// By default, the subject field is not included in tags as it may contain sensitive data.
+// Use WithSubject() option to include it.
+func TraceWrapCloudEventsHandler(originalHandler func(context.Context, cloudevents.Event) error, resourceName string, opts ...Option) func(context.Context, cloudevents.Event) error {
+	cfg := defaultConfig()
+	for _, opt := range opts {
+		opt(cfg)
+	}
+
+	return func(ctx context.Context, event cloudevents.Event) (err error) {
+		parentSpanCtx, _ := ExtractTraceContext(event)
+
+		spanOpts := []tracer.StartSpanOption{
+			tracer.ResourceName(resourceName),
+			tracer.SpanType(ext.SpanTypeMessageConsumer),
+			tracer.Tag("event.id", event.ID()),
+			tracer.Tag("event.type", event.Type()),
+			tracer.Tag("event.source", event.Source()),
+			tracer.Tag("message_size", len(event.Data())),
+			tracer.Tag(ext.Component, "cloudevents"),
+			tracer.Tag(ext.SpanKind, ext.SpanKindConsumer),
+			tracer.ChildOf(parentSpanCtx),
+		}
+
+		if cfg.includeSubject && event.Subject() != "" {
+			spanOpts = append(spanOpts, tracer.Tag("event.subject", event.Subject()))
+		}
+
+		if parentSpanCtx != nil && parentSpanCtx.SpanLinks() != nil {
+			spanOpts = append(spanOpts, tracer.WithSpanLinks(parentSpanCtx.SpanLinks()))
+		}
+
+		span, ctx := tracer.StartSpanFromContext(ctx, "cloudevents.receive", spanOpts...)
+		defer span.Finish(tracer.WithError(err))
+
+		return originalHandler(ctx, event)
+	}
+}
+
+// InjectTraceContext injects Datadog trace context into a CloudEvent's extensions.
+// This should be called by publishers to propagate trace context across service boundaries.
+//
+// Example:
+//
+//	span, ctx := tracer.StartSpanFromContext(ctx, "publish.event")
+//	defer span.Finish()
+//	event := cloudevents.NewEvent()
+//	if err := InjectTraceContext(span.Context(), &event); err != nil {
+//	    return err
+//	}
+func InjectTraceContext(spanCtx *tracer.SpanContext, e *cloudevents.Event) error {
```

Comment:
```suggestion
func Inject(spanCtx *tracer.SpanContext, e *cloudevents.Event) error {
```

Aligning with our usual semantics.

Reply:
@rarguelloF @darccio thanks a lot for the inputs, I actually think the NewContextCarrier is very idiomatic, I can definitelly implement that.

In the past 2 days I also created a TraceClient that wraps the Cloudevents.Client and the Send function to properly wrap tracing around the message, I will pop it in taking the comments into consideration